### ISSUE TITLE
Implement vrDisplay.getLayers

### ISSUE THREAD

**Issue by spite** [spite/WebVR-Extension#18]
Return an array of VRLayer (TBD)

**Comment by AVGP** [comment]
That's also blocked by the wrapping rewrite, huh?

**Comment by spite** [comment]
Yes, on hold. Not really blocked, i haven't had time to add it yet (it has to work together with VRDisplay. requestPresent).

**Comment by AVGP** [comment]
I was just wondering how to go about that.
I see three options in the current code:
- Store the layers in a variable that is local to the module to hide it
  from the outside
- Store the layers in a "private" property, such as `__layers` on the
  prototype of `VRDisplay`
- wrap the object to have a private variable in the closure for VRDisplay

Or am I missing a way to store the information?

Am 21.10.2016 2:58 vorm. schrieb "Jaume Sanchez" notifications@github.com:

**Comment by spite** [comment]
All these options are valid, really. 

The quid here is converting from the `canvas` elements passed to `requestPresent` into a `VRLayer` object: that is, populate `leftBounds` and `rightBounds` correctly.

**Comment by AVGP** [comment]
Fun thing: I tried with the Vive and the latest Chromium build and `leftBounds` and `rightBounds` are set to `null`. So for emulation that's a get-out-of-jail-free card, but I wonder if that implementation is deliberately that way.

**Comment by spite** [comment]
That makes sense. I have a feeling the concept of layers will be useful and very powerful in future versions of the implementation, but for now, limited to a single canvas, it's pretty straightforward.

**Comment by spite** [comment]
Implemented on https://github.com/spite/WebVR-Extension/pull/19#issuecomment-256964223 and deployed on 1.1.2. Thanks!
